mcstatus/protocol/connection.py

```diff
@@ -302,6 +310,10 @@ def read_ulong(self) -> int:
         """Return ``0 - 18446744073709551613 (2 ** 64 - 1)``. Read 8 bytes."""
         return self._unpack("Q", self.read(8))
 
+    def read_bool(self) -> bool:
+        """Return `True` or `False`. Read 1 byte."""
+        return self._unpack("?", self.read(1)) == 1
```

Comment:
`struct.unpack(">?", struct.pack(">?", True))` gives `(True,)`, so why not compare it to `True`?

```suggestion
        return self._unpack("?", self.read(1)) == True
```

The same for the async version.

---

This works now because `True == 1` (`bool` is a subclass of `int`).

Reply:
Why compare at all? Why not just `self._unpack("?", self.read(1))[0]`?

Reply:
Basically because funny things with type annotations. We've annotated _unpack saying that it always returns an integer, even though in reality when we use "?" it actually returns a boolean value. The real question is should we fix _unpack to use overrides, or should we just pretend everything is an integer and just do kind of silly things like this to make mypy happy

Reply:
Yeah, I like the cast more in this case.

Ultimately, overloads would be the best solution, but we'll probably migrate this library to eventually use `mcproto` for this low level networking, which already has this implemented properly (and indeed, that's done with overloads).